Build failures with mtl-2.3

When I build this package with the `mtl-2.3` release I get this failure:

```
src/Data/Aeson/Commit.hs:64:40: error:
    Not in scope: type constructor or class ‘ExceptT’
   |
64 | newtype Commit a = Commit {unCommit :: ExceptT [Parser Void] Parser a}
   |                                        ^^^^^^^
```

This build failure is due to https://github.com/haskell/mtl/issues/117 which will be fixed in a follow-up release of `mtl`.

Building with a version of `mtl` that includes a fix to that issue, I'm getting

```
src/Data/Aeson/Commit.hs:73:3: error:
    Variable not in scope:
      lift :: Parser b -> ExceptT [Parser Void] Parser b
   |
73 |   lift $ post a
   |   ^^^^
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Changelog
 
+## [1.5]
+### [Changed]
+- [#15] Depend on transformers instead of mtl, to sidestep current export issues in mtl 2.3
+
 ## [1.4]
 ### [Added]
 - Nix support

--- a/aeson-commit.cabal
+++ b/aeson-commit.cabal
@@ -30,10 +30,10 @@ library
   default-language: Haskell2010
   ghc-options:      -Wall -Wno-name-shadowing
   build-depends:
-      aeson  >=1.4  && <2.1
-    , base   >=4.10 && <5
-    , mtl    >=2.2  && <3
-    , text   >=1.2  && <2
+      aeson         >=1.4  && <2.1
+    , base          >=4.10 && <5
+    , text          >=1.2  && <2
+    , transformers  >=0.5  && <0.7
 
 test-suite tasty
   type:             exitcode-stdio-1.0

--- a/aeson-commit.cabal
+++ b/aeson-commit.cabal
@@ -1,6 +1,6 @@
 cabal-version:      >=1.10
 name:               aeson-commit
-version:            1.4
+version:            1.5
 license:            BSD3
 copyright:          2020 Cross Compass Ltd.
 maintainer:         Jonas Carpay <jonascarpay@gmail.com>

--- a/src/Data/Aeson/Commit.hs
+++ b/src/Data/Aeson/Commit.hs
@@ -41,9 +41,11 @@ module Data.Aeson.Commit
 where
 
 import Control.Applicative (Alternative (..))
-import Control.Monad.Except
+import Control.Monad.Trans.Class (lift)
+import Control.Monad.Trans.Except
 import Data.Aeson.Types
 import Data.Void (Void, absurd)
+
 #if MIN_VERSION_aeson(2,0,0)
 import qualified Data.Aeson.Key as Key
 #else

